### PR TITLE
Add rule S1117 (former S1524) (based on 'no-shadow') for TS/JS

### LINKED ISSUE #1417: Add rule S1117 (former S1524) (based on `no-shadow`) for TS/JS
This rule in JS has key S1117, in TS S1524. RSPEC-1524 is deprecated. So we should declare `S1524` as deprecated key.

### PR DESCRIPTION
**Pull request by vilchik-elena**
Fixes #1417

### Changes for JS:

* there are new issues for function expression name `hello = function hello() {}`
* there are new issues for variable/parameter shadowing imported symbol
* there are new issues for class names shadowing
* and some missing issue related to `var` variable declarations in sibling scopes.

### CODE REVIEW

File: its/ruling/src/test/expected/js/js-project/javascript-S1117.json [hunk 1]
```diff
@@ -129,6 +129,10 @@
 'js-project:babylon/src/plugins/typescript.js':[
 1728,
 ],
+'js-project:babylon/test/helpers/runFixtureTests.js':[
+59,
```
Comment (quentin-jaquier-sonarsource) [inline its/ruling/src/test/expected/js/js-project/javascript-S1117.json]: Another example of variable shadowing an import.

File: its/ruling/src/test/expected/js/js-project/javascript-S1117.json [hunk 2]
```diff
@@ -728,11 +886,19 @@
...
 'js-project:ecmascript6/react-router/modules/__tests__/Router-test.js':[
+91,
+108,
 149,
+253,
+259,
 331,
 ],
 'js-project:ecmascript6/react-router/modules/__tests__/_bc-Router-test.js':[
+91,
```
Comment (quentin-jaquier-sonarsource) [inline its/ruling/src/test/expected/js/js-project/javascript-S1117.json]: Class shadowed are also reported?

File: its/ruling/src/test/expected/js/js-project/javascript-S1117.json [hunk 3]
```diff
@@ -1104,6 +1268,9 @@
 378,
 408,
 ],
+'js-project:jest/packages/jest-jasmine2/src/jest_expect.js':[
+55,
```
Comment (quentin-jaquier-sonarsource) [inline its/ruling/src/test/expected/js/js-project/javascript-S1117.json]: New issues: variable shadowing an import.

File: its/ruling/src/test/expected/js/js-project/javascript-S1117.json (hunk 2 above)
Comment (vilchik-elena) [reply its/ruling/src/test/expected/js/js-project/javascript-S1117.json]: indeed, I will add it to description

**Comment by vilchik-elena** [comment]
@quentin-jaquier-sonarsource thanks for these findings. About RSPEC I think it's generic enough to keep it as it is.

**Pull request merged**